### PR TITLE
Add initial CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

Adding initial CONTRIBUTING.md, based on images, cri-o and skopeo.

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
@nalind @rhatdan PTAL

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+Use your real name (sorry, no pseudonyms or anonymous contributions.)
+
+If you set your `user.name` and `user.email` git configs, you can sign your
+commit automatically with `git commit -s`.
+
+## Communications
+
+For general questions, or discussions, please use the
+IRC group on `irc.freenode.net` called `cri-o`
```
Comment (runcom) [inline CONTRIBUTING.md]: The channel is `#container-projects`

Comment (TomSweeneyRedHat) [reply CONTRIBUTING.md]: I can change that, but containers/storage points to cri-o, should I put that one to container-projects too?

Comment (runcom) [reply CONTRIBUTING.md]: Yes, please update c/storage as well

**Comment by runcom** [comment]
LGTM

**Comment by rhatdan** [comment]
@mtrmac  Could you approve this?

**Comment by TomSweeneyRedHat** [comment]
@runcom @mtrmac  I think I've addressed @runcom's concerns by changing the IRC channel.  Please merge this or let me know what I need to do to get it ready to go.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+Regardless of the type of PR, all PRs should include:
+* well documented code changes
+* additional testcases. Ideally, they should fail w/o your code change applied
+* documentation changes
+
+Squash your commits into logical pieces of work that might want to be reviewed
+separate from the rest of the PRs. But, squashing down to just one commit is ok
+too since in the end the entire PR will be reviewed anyway. When in doubt,
+squash.
```
Comment (mtrmac) [inline CONTRIBUTING.md]: (? I’m more the “when in doubt, split it” camp; for me, reviewing a hundred 10-line commits, as long as each commit is a logical, self-contained step, is much easier than reviewing a 1000-line PR which only outlines the broad goals.  I don’t care care about this _too_ strongly, but this seems surprising to me.  @runcom is this an intentional preference, or just something copy&pasted over 50 projects?)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+If you find a new issue with the project we'd love to hear about it! The most
+important aspect of a bug report is that it includes enough information for
+us to reproduce it. So, please include as much detail as possible and try
+to remove the extra stuff that doesn't really relate to the issue itself.
+The easier it is for us to reproduce it, the faster it'll be fixed!
+
+Please don't include any private/sensitive information in your issue!
+
+## Submitting Pull Requests
```
Comment (mtrmac) [inline CONTRIBUTING.md]: Should this refer to “the third paragraph” (we really should make that a named section) of https://github.com/projectatomic/skopeo/blob/master/README.md#dependencies-management as well?

File: CONTRIBUTING.md (hunk 2 above)
Comment (mtrmac) [reply CONTRIBUTING.md]: (OTOH some people tend create many-commit PRs where each commit is just a random collection of unrelated edits; those are useless, sure. But GitHub allows reviewing the PR as a whole, so even this does not hurt reviewing—only `git bisect`.)

Comment (TomSweeneyRedHat) [reply CONTRIBUTING.md]: FWIW, Dan grabbed this doc (I think from skopeo) and dropped it into CRI-O with a few changes to point to the right project/issue links.  I've done the same for this one and a couple others.  I'm fine with stating whatever, but if we need to change this one, we ought to change the other 4 or 5 too.

But FWIW, I totally concur with the smaller is better commit.  kpod recently had a 5K line submit and that was a beast.

Comment (mtrmac) [reply CONTRIBUTING.md]: > I'm fine with stating whatever, but if we need to change this one, we ought to change the other 4 or 5 too.

\*shrug\* I have a preference, but I’m not _quite_ as self-assured to insist that everything must be done that way :)

Let’s merge this, and I can submit a follow-up PR to change this bit.

**Comment by mtrmac** [comment]
:+1: ; please rebase.

**Comment by TomSweeneyRedHat** [comment]
Rebased!  But I was surprised that it needed it again.  "shrug"

**Pull request merged**